pkg/operator/ceph/cluster/watcher.go

```diff
@@ -438,13 +438,18 @@ func (c *clientCluster) fenceCephFSVolume(
 		return fmt.Errorf("failed to list watchers for cephfs pool/subvoumeName %s/%s. %v", cephFSPV.Spec.CSI.VolumeAttributes["pool"], cephFSPV.Spec.CSI.VolumeAttributes["subvolumeName"], err)
 	}
 	ips, err := cephFSMDSClientMarshal(buf, cephFSPV)
-	if err != nil || ips == nil {
-		return fmt.Errorf("failed to unmarshal cephfs mds  output. %v", err)
+	if err != nil {
+		return pkgerror.Wrapf(err, "failed to unmarshal cephfs mds output")
+	}
+
+	if len(ips) == 0 {
+		logger.Infof("no active mds clients found for cephfs volume %q", cephFSPV.Name)
+		return nil
```

Comment:
By returning `nil`, shouldn't this mean the fencing was successful? But the fencing is being skipped. How will the fencing be retried in the future if we are returning nil? Don't we need to return an error so the fencing will try again?

Reply:
Yes true, we are doing the same for RBD https://github.com/rook/rook/blob/64cf8f22e5a633d514b3034511b97a5e7c893e24/pkg/operator/ceph/cluster/watcher.go#L404-L409, if we don't any client using that subvolume/rbdimage we are assuming that no client is connected and no need for blocklisting.

Reply:
For both rbd and cephfs, the calling method is in a `for` loop. If `nil` is returned, the `for` loop breaks out. In the case of no IPs found, should the `for` loop keep iterating through other PVs instead of aborting? Or all PVs will have the same result of no IPs found?

Reply:
No, you are correct we should try for other PVC, added new comment for introducing new check. PTAL